logFilter.js

```diff
@@ -12,12 +12,22 @@ function filterLogsByEnv(logs, env) {
 
 function filterLogsOnDate(logs, dateFilter) {
   const filteredLogs = []
-  const parsedDateFilter = new Date(dateFilter)
-
-  logs.forEach((log) => {
-    const timestampDate = new Date(log.timestamp)
-
-    if (parsedDateFilter.toDateString() === timestampDate.toDateString()) {
+  let parsedDateFilter = new Date(dateFilter)
+  parsedDateFilter = new Date(
+    parsedDateFilter.getFullYear(),
+    parsedDateFilter.getUTCDate(),
+    parsedDateFilter.getUTCDay()
+  )
+
+  logs.forEach(log => {
+    let timestampDate = new Date(log.timestamp)
+    timestampDate = new Date(
+      timestampDate.getFullYear(),
+      timestampDate.getUTCDate(),
+      timestampDate.getUTCDay()
+    )
+
+    if (parsedDateFilter.toString() === timestampDate.toString()) {
```

Comment:
The reason this works is because both dates you created are set to midnight UTC on 2021-05-04. `toString()` will convert both to local time which will be 8pm on 2021-05-03. It works because the date code above ends up setting them both to midnight. This ends up solving the time difference we saw before when we converted midnight and got the day before and converted a later time and didn't go back a day but it only works because you dropped the time.

A better solution would have been to leave both dates in UTC as they were originally and instead of using a function like `toDateString` or `toString` which both convert to local time, you could have used a series of UTC functions to make sure they happened on the say exact date.

```javascript
if(parsedDateFilter.getUTCFullYear() === timestampDate.getUTCFullYear() &&
  parsedDateFilter.getUTCMonth() === timestampDate.getUTCMonth() &&
  parsedDateFilter.getUTCDate() === timestampDate.getUTCDate()) {
```